k4Reco/DDPlanarDigi/components/DDPlanarDigi.h

```diff
@@ -0,0 +1,138 @@
+
+  StatusCode initialize() override;
+  StatusCode finalize() override;
+
+  std::tuple<TrackerHitPlaneColl, Association> operator()(const SimTrackerHitCollection& simTrackerHits,
+                                                          const Header&                  headers) const override;
+
+private:
+  Gaudi::Property<std::string>        m_subDetName{this, "SubDetectorName", "VXD", "Name of the subdetector"};
+  Gaudi::Property<bool>               m_isStrip{this, "IsStrip", false, "Whether the hits are 1D strip hits"};
+  Gaudi::Property<std::vector<float>> m_resULayer{
+      this, "ResolutionU", {0.004}, "Resolution in the direction of u; either one per layer or one for all layers"};
+  Gaudi::Property<std::vector<float>> m_resVLayer{
+      this, "ResolutionV", {0.004}, "Resolution in the direction of v; either one per layer or one for all layers"};
+  Gaudi::Property<std::vector<float>> m_resTLayer{
+      this,
+      "ResolutionT",
+      {0.004},
+      "Resolution in the direction of t; either one per layer or one for all layers. If the single entry is negative, "
+      "disable time smearing. "};
+  Gaudi::Property<bool>   m_forceHitsOntoSurface{this, "ForceHitsOntoSurface", false,
+                                               "Project hits onto the surfoce in case they are not yet on the surface"};
+  Gaudi::Property<double> m_minEnergy{this, "MinEnergy", 0.0, "Minimum energy (GeV) of SimTrackerHit to be digitized"};
+
+  Gaudi::Property<bool> m_useTimeWindow{
+      this, "UseTimeWindow", false,
+      "Only accept hits with time (after smearing) within the specified time window (default: false)"};
+  Gaudi::Property<bool> m_correctTimesForPropagation{
+      this, "CorrectTimesForPropagation", false,
+      "Correct hit time for the propagation: radial distance/c (default: false)"};
+  Gaudi::Property<std::vector<float>> m_timeWindowMin{
+      this, "TimeWindowMin", {-1e9}, "Minimum time (ns) of SimTrackerHit to be digitized"};
+  Gaudi::Property<std::vector<float>> m_timeWindowMax{
+      this, "TimeWindowMax", {1e9}, "Maximum time (ns) of SimTrackerHit to be digitized"};
+  Gaudi::Property<std::string> m_encodingStringVariable{
+      this, "EncodingStringParameterName", "GlobalTrackerReadoutID",
+      "The name of the DD4hep constant that contains the Encoding string for tracking detectors"};
+  Gaudi::Property<std::string> m_geoSvcName{this, "GeoSvcName", "GeoSvc", "The name of the GeoSvc instance"};
+  Gaudi::Property<int> m_maxTries{this, "MaxTries", 10, "Maximum number of tries to find a valid surface for a hit"};
+
+  Gaudi::Property<std::string> m_outputFileName{this, "OutputFileName", "planar_digi_histograms.root", "Output file name for the histograms"};
+
+  const dd4hep::rec::SurfaceMap* surfaceMap;
+  // std::vector<std::unique_ptr<Gaudi::Accumulators::Histogram<1>>> m_histograms;
+  std::vector<std::unique_ptr<Gaudi::Accumulators::Histogram<1>>> m_histograms;
```

Comment:
This could be an array, right? The size is effectively static. The only downside is that we would need to make the `enum` from the implementation a member of the class if we don't want to introduce a "magic" size.

Reply:
Yep, my idea was to move to `HistogramArray` which exists in Gaudi but it's not as flexible as we would need here and I wasn't completely happy with it so I just changed it to an array. Also these histograms do not store stats so if you browse them the values of the mean and std. dev. may be wrong... https://gitlab.cern.ch/gaudi/Gaudi/-/issues/281